### ISSUE TITLE
Add ability for manageJSS to skip re-generating existing styles [fast-jss-manager-react]

### ISSUE THREAD

**Issue by k1mby** [microsoft/fast-dna#2218]
For server side rendering we are able to generate the css with matching classnames and pass it to the client. 

It would be nice to then be able to just use the existing styles and avoid re-generating them on the client.

**Comment by nicholasrice** [comment]
I agree, this is a feature we're looking into but it isn't very straight forward. There is currently (to my knowledge) no utility in JSS for hydrating a JSS style object with a style DOM element. 

One solution I'm playing with is adding the ability to defer or prevent attachment of JSS generated stylesheets on hydration, meaning the work to add a style element wouldn't happen until a second render. While this would speed up initial hydration performance, it does seems somewhat fragile and puts onus on the app author to determine when hydration is done.

If you have any proposals I would love to hear them.

**Comment by chrisdholt** [comment]
closed with #3517
